libs/core/src/periodic_runnable.cpp

```diff
@@ -0,0 +1,44 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/periodic_runnable.hpp"
+
+namespace fetch {
+namespace core {
+
+PeriodicRunnable::PeriodicRunnable(Duration const &period)
+  : last_executed_{Clock::now()}
+  , interval_{period}
+{
+}
+
+bool PeriodicRunnable::IsReadyToExecute() const
```

Comment:
We might want to make runnable sortable according to next expected runtime and put time in a priority queue.

Reply:
Yes this would require an interface update but would be a good idea. I am inclined to see if we run into any issues because of the ordering of the jobs in the reactor. For the most part we only use it for cases where the deadline is strictly a hint rather than a hard request.